### PR TITLE
STI

### PR DESCRIPTION
**Pull request by djleo95**

### CODE REVIEW

File: app/models/human.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Human < ApplicationRecord
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+  format: {with: VALID_EMAIL_REGEX},
```
Comment (yeunuthongminh) [inline app/models/human.rb]: indent

File: app/models/human.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class Human < ApplicationRecord
+  validates :name, presence: true, length: {maximum: 50}
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
+
+  self.inheritance_column = :role
+
+  scope :guests, -> { where(role: "Guest")}
```
Comment (yeunuthongminh) [inline app/models/human.rb]: REMOVE ()

File: app/models/human.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  validates :email, presence: true, length: {maximum: 255},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
+
+  self.inheritance_column = :role
+
+  scope :guests, -> { where role: "Guest"}
+  scope :admins, -> { where role: "Admin"}
+
+  def self.role
```
Comment (yeunuthongminh) [inline app/models/human.rb]: dùng class method đi

File: app/models/human.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  scope :guests, -> { where role: "Guest"}
+  scope :admins, -> { where role: "Admin"}
+
+  class << self
+    def role
+      %w(Guest Admin)
+    end
+  end
+
+  def self.digest(string)
```
Comment (yeunuthongminh) [inline app/models/human.rb]: class method

File: app/assets/javascripts/custom.js [hunk 5]
```diff
@@ -110,3 +110,6 @@ $(document).ready(ready);
 $(document).on('click', '.edit_pass', function(){
   $('#edit_pass_show').toggleClass('edit_pass_content_hide', 'edit_pass_content_show');
 });
+$(document).ready(setTimeout(function(){
+  $(".alert, .danger, .success, .error, .notice, .info" ).fadeOut(3000);
```
Comment (dat-hedspi) [inline app/assets/javascripts/custom.js]: dùng single quote

File: app/models/human.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  scope :guests, -> { where role: "Guest"}
+  scope :admins, -> { where role: "Admin"}
+
+  class << self
+    def role
+      %w(Guest Admin)
+    end
+
+    def digest(string)
```
Comment (dat-hedspi) [inline app/models/human.rb]: bỏ ()

File: app/models/human.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+        BCrypt::Engine.cost
+      BCrypt::Password.create(string, cost: cost)
```
Comment (dat-hedspi) [inline app/models/human.rb]: bỏ ()

File: app/models/human.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+class Human < ApplicationRecord
+  validates :name, presence: true, length: {maximum: 50}
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (yeunuthongminh) [inline app/models/human.rb]: indent

**Comment by yeunuthongminh** [comment]
OK

**Comment by djleo95** [comment]
Framgia CI Report Tool 0.2.1

------------------------------------------
Running Test
------------------------------------------

[+] Running:  bundle-audit check --update | tee .framgia-ci-reports/bundle-audit.txt
Updating ruby-advisory-db ...
From https://github.com/rubysec/ruby-advisory-db
 * branch            master     -> FETCH_HEAD
Already up-to-date.
Updated ruby-advisory-db
ruby-advisory-db: 284 advisories
Name: nokogiri
Version: 1.7.0.1
Advisory: CVE-2016-4658
Criticality: Unknown
URL: https://github.com/sparklemotion/nokogiri/issues/1615
Title: Nokogiri gem contains several vulnerabilities in libxml2 and libxslt
Solution: upgrade to >= 1.7.1

Vulnerabilities found!
[+] Running:  rspec --format html --out .framgia-ci-reports/rspec.html spec/
sh: 1: rspec: not found
[+] Running:  scss-lint --require=scss_lint_reporter_checkstyle --format=Checkstyle --out=.framgia-ci-reports/scss-lint.xml app/assets/stylesheets/
[+] Running:  rake eslint:run_all --format=checkstyle --output-file=.framgia-ci-reports/eslint.xml app/
invalid option: --format=checkstyle
[+] Running:  rails_best_practices --format html --output-file .framgia-ci-reports/rails_best_practices.html app/
Source Code: |========================================================================================================================================|
[+] Running:  reek --format html > .framgia-ci-reports/reek.html
[+] Running:  bundle exec rubocop --require rubocop/formatter/checkstyle_formatter --format RuboCop::Formatter::CheckstyleFormatter --no-color --rails --out .framgia-ci-reports/rubocop.xml app/ lib/
/home/justakiss/Ruby/project2/restaurant_manager_2/.rubocop.yml: Style/BlockLength has the wrong namespace - should be Metrics
/home/justakiss/Ruby/project2/restaurant_manager_2/.rubocop_enabled.yml: Lint/Eval has the wrong namespace - should be Security
Warning: unrecognized cop Style/DeprecatedHashMethods found in /home/justakiss/Ruby/project2/restaurant_manager_2/.rubocop_enabled.yml
/home/justakiss/Ruby/project2/restaurant_manager_2/.rubocop.yml - AllCops/Excludes was renamed to AllCops/Exclude
Warning: unrecognized cop Style/DeprecatedHashMethods found in /home/justakiss/Ruby/project2/restaurant_manager_2/.rubocop.yml
Warning: unrecognized parameter Style/TrailingCommaInLiteral:SupportedStyles found in /home/justakiss/Ruby/project2/restaurant_manager_2/.rubocop.yml
Warning: unrecognized parameter Lint/EndAlignment:SupportedStyles found in /home/justakiss/Ruby/project2/restaurant_manager_2/.rubocop.yml
Warning: unrecognized parameter Lint/DefEndAlignment:SupportedStyles found in /home/justakiss/Ruby/project2/restaurant_manager_2/.rubocop.yml
[+] Running:  brakeman -o .framgia-ci-reports/brakeman.html -o .framgia-ci-reports/brakeman.json
Loading scanner...
Processing application in /home/justakiss/Ruby/project2/restaurant_manager_2
Processing gems...
[Notice] Detected Rails 5 application
Processing configuration...
[Notice] Escaping HTML by default
Parsing files...
Processing initializers...
Processing libs...ed
Processing routes...          
Processing templates...       
Processing data flow in templates...
Processing models...          
Processing controllers...     
Processing data flow in controllers...
Indexing call sites...        
Running checks in parallel...
 - CheckBasicAuth
 - CheckBasicAuthTimingAttack
 - CheckCrossSiteScripting
 - CheckContentTag
 - CheckCreateWith
 - CheckDefaultRoutes
 - CheckDeserialize
 - CheckDetailedExceptions
 - CheckDigestDoS
 - CheckDynamicFinders
 - CheckEscapeFunction
 - CheckEvaluation
 - CheckExecute
 - CheckFileAccess
 - CheckFileDisclosure
 - CheckFilterSkipping
 - CheckForgerySetting
 - CheckHeaderDoS
 - CheckI18nXSS
 - CheckJRubyXML
 - CheckJSONEncoding
 - CheckJSONParsing
 - CheckLinkTo
 - CheckLinkToHref
 - CheckMailTo
 - CheckMassAssignment
 - CheckMimeTypeDoS
 - CheckModelAttrAccessible
 - CheckModelAttributes
 - CheckModelSerialize
 - CheckNestedAttributes
 - CheckNestedAttributesBypass
 - CheckNumberToCurrency
 - CheckQuoteTableName
 - CheckRedirect
 - CheckRegexDoS
 - CheckRender
 - CheckRenderDoS
 - CheckRenderInline
 - CheckResponseSplitting
 - CheckRouteDoS
 - CheckSafeBufferManipulation
 - CheckSanitizeMethods
 - CheckSelectTag
 - CheckSelectVulnerability
 - CheckSend
 - CheckSendFile
 - CheckSessionManipulation
 - CheckSessionSettings
 - CheckSimpleFormat
 - CheckSingleQuotes
 - CheckSkipBeforeFilter
 - CheckSQL
 - CheckSQLCVEs
 - CheckSSLVerify
 - CheckStripTags
 - CheckSymbolDoSCVE
 - CheckTranslateBug
 - CheckUnsafeReflection
 - CheckValidationRegex
 - CheckWithoutProtection
 - CheckXMLDoS
 - CheckYAMLParsing
Checks finished, collecting results...
Generating report...
Report saved in '.framgia-ci-reports/brakeman.html'
Report saved in '.framgia-ci-reports/brakeman.json'

------------------------------------------
[x] Build Fail!
------------------------------------------

[o] bundle-audit: success
[x] rspec: failed
[x] scss-lint: failed
[x] eslint: failed
[x] rails_best_practices: failed
[x] reek: failed
[o] rubocop: success
[o] brakeman: success

**Pull request merged**